bin/check-aggregate-output.rb

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/env ruby
+#
+# Check Aggregate Output
+# ===
+# This check is very similar to check-aggregate.rb, it checks the Sensu aggregate
+# api and returns Critical / Warning / OK based on how many aggregate results
+# match the options you configure.
+#
+# Authors
+# ===
+# AJ Bourg, @ajbourg
+# Sean Porter, @portertech
+#
+# Copyright 2017
+#
+# Released under the same terms as Sensu (the MIT license); see
+# LICENSE for details.
+
+require 'sensu-plugin/check/cli'
+require 'rest-client'
+require 'json'
+
+class CheckAggregateOutput < Sensu::Plugin::Check::CLI
+  option :api,
+         short: '-a URL',
+         long: '--api URL',
+         description: 'Sensu API URL',
+         default: if ENV['SENSU_API']
+                    ENV['SENSU_API'] + ':4567'
+                  elsif ENV['SENSU_API_URL']
+                    ENV['SENSU_API_URL']
+                  else
+                    'http://localhost:4567'
+                  end
+
+  option :insecure,
+         short: '-k',
+         boolean: true,
+         description: 'Enabling insecure connections',
+         default: false
+
+  option :user,
+         short: '-u USER',
+         long: '--user USER',
+         description: 'Sensu API USER'
+
+  option :password,
+         short: '-p PASSWORD',
+         long: '--password PASSWORD',
+         description: 'Sensu API PASSWORD'
+
+  option :timeout,
+         short: '-t SECONDS',
+         long: '--timeout SECONDS',
+         description: 'Sensu API connection timeout in SECONDS',
+         proc: proc(&:to_i),
+         default: 30
+
+  option :check,
+         short: '-c CHECK',
+         long: '--check CHECK',
+         description: 'Aggregate CHECK name',
+         required: true
+
+  option :age,
+         short: '-A SECONDS',
+         long: '--age SECONDS',
+         description: 'Minimum aggregate age in SECONDS, time since check request issued',
+         default: 30,
+         proc: proc(&:to_i)
+
+  option :message,
+         short: '-M MESSAGE',
+         long: '--message MESSAGE',
+         description: 'A custom error MESSAGE'
+
+  option :collect_output,
+         short: '-o',
+         long: '--output',
+         boolean: true,
+         description: 'Collects all non-ok outputs',
+         default: true
+
+  option :warning,
```

Comment:
I read through this again and maybe it does make sense as is so that you can monitor both. For example you might want to alert on one handler definition for SLA threshold but another for do early detection. This really is expanding on the differences between severity and urgency. I do think that it would still be better served as 2 seperate check definitions as it provides more control and clarity to operators.